src/control.cc

```diff
@@ -160,41 +165,41 @@ bool control_conn_t::process_find_load(int pktType)
     
     service_record * record = nullptr;
     
-    string serviceName = rbuf.extract_string(3, svcSize);
+    string service_name = rbuf.extract_string(3, srvname_len);
 
     // Clear the packet from the buffer
     rbuf.consume(chklen);
     chklen = 0;
     
-    char fail_code = DINIT_RP_NOSERVICE;
+    cp_rply fail_code = cp_rply::NOSERVICE;
 
-    if (pktType == DINIT_CP_LOADSERVICE) {
+    if (pktType == cp_cmd::LOADSERVICE) {
         // LOADSERVICE
         try {
-            record = services->load_service(serviceName.c_str());
+            record = services->load_service(service_name.c_str());
         }
         catch (service_description_exc &sdexc) {
             log_service_load_failure(sdexc);
-            fail_code = DINIT_RP_SERVICE_DESC_ERR;
+            fail_code = cp_rply::SERVICE_DESC_ERR;
         }
         catch (service_not_found &snf) {
             log(loglevel_t::ERROR, "Could not load service ", snf.service_name, ": ",
                     snf.exc_description);
-            // fail_code = DINIT_RP_NOSERVICE;   (already set)
+            // fail_code = cp_rply::NOSERVICE;   (already set)
         }
         catch (service_load_exc &slexc) {
             log(loglevel_t::ERROR, "Could not load service ", slexc.service_name, ": ",
                     slexc.exc_description);
-            fail_code = DINIT_RP_SERVICE_LOAD_ERR;
+            fail_code = cp_rply::SERVICE_LOAD_ERR;
         }
     }
     else {
         // FINDSERVICE
-        record = services->find_service(serviceName.c_str());
+        record = services->find_service(service_name.c_str());
     }
     
     if (record == nullptr) {
-        std::vector<char> rp_buf = { fail_code };
+        std::vector<char> rp_buf = { (cp_rply_t)fail_code };
```

Comment:
(This one is causing build failure on Mac, clang is less lenient than gcc)

Reply:
The underlying Problem seems to be, that when variables in the initilizer list are non static and the types are not the same then there is narrowing involved (which I don't fully understand). So casting to (char) works, but casting to (cp_rply_t)/(unsigned char) does not, as it's not clear if the value fits into char. So I've opted for casting to (char) in this case.